lib/core.js

```diff
@@ -64,7 +64,7 @@ function Core(options) {
     // Alternatively, if the connectionOptions includes token information then we will use that
     // instead of the username/password
 
-    if (connectionOptions) {
+    if (connectionOptions == null) {
```

Comment:
it will handle also undefined? maybe it should be `!connectionOptions`?

Reply:
It handle undefined using '==' instead of '==='